src/server/api/controllers/candidates.controller.js

```diff
@@ -35,11 +35,21 @@ const createCandidate = async (userId, boardId, newCandidate) => {
   if (typeof newCandidate.name !== 'string') {
     throw new IncorrectEntryError(`Candidate name should be string`);
   }
-  const createNewCandidate = await knex('candidates').insert({
-    boardId,
-    name: newCandidate.name,
-    isBlocked: false,
-  });
+  await checkUserRole({ userId, boardId });
+  const createNewCandidate = await knex('candidates')
+    .insert({
+      boardId,
+      name: newCandidate.name,
+      isBlocked: false,
+    })
+    .then(function (response) {
+      return knex('ballots').insert({
+        boardId,
+        userId,
+        candidateId: response[0],
+        rank: -1,
+      });
```

Comment:
I don't think you can use `.then` (promise-based method) since you're using async. Also, when the owner adds a new candidate, it needs to be added to every member's ballot, not just the owner themself.

Reply:
Hi Moritz, Thanks for the clarification
I tried to add ballots for every member in the board now. Can you please take a look at it